### PR TITLE
Doc updates - Troubleshooting and other updates. 

### PR DESCRIPTION
**Pull request by RadhakrishnaJ**
Main Readme - 
Added Troubleshooting section in main readme.

Download and setup - 
Corrected resources reference link used for Content Trust in Docker. 
Added statement to clarify need for referring autopass license server setup pdf.  

Troubleshooting - 
Added Troubleshooting in documents.
Added instructions to resolve 6002 Error. 

Examples - 
Added clarity for APLS IP.

FAQs - 
Updated FAQs to install custom packages for Machine Learning.

### CODE REVIEW

File: docs/FAQ.md [hunk 1]
```diff
@@ -130,8 +130,16 @@ We support the following basic packages required for ML - numpy, scipy, matplotl
 
 ## Can one add new python packages in Swarm Learning?
 
-Yes. If user needs additional packages, he can spawn a shell in the Swarm Learning (SL) container and add the required packages. He can also
-do a docker commit to persist this across restarts.
+Yes. For any additional packages support to machine learning process, user has to do following.
```
Comment (suresh-ls) [inline docs/FAQ.md:133]: Reword - "additional support packages"..

File: docs/Troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Troubleshooting provides solutions to commonly observed issues during Swarm set up and execution.
+
+
+### 1. Error code : 6002
+####    Error message: Unable to connect to server. Server might be wrongly configured or down. #### 
+####    Custom message: Error in communicating with server https://HOST_SYSTEM_IP:5814 (default port) #### 
+
+#### Problem Description 
+Error code: 6002, as shown in below screenshot happens when running SL Node without starting APLS server and License setup.
```
Comment (suresh-ls) [inline docs/Troubleshooting.md:11]: Better to more specific 'SL is not able to connect to the APLS server'

File: docs/Troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+Fix for above issue involves two things.
+1.	Running AutoPass License Server (APLS) 
+2.	Setting up Swarm License
+
+##### 1. Running AutoPass License Server (APLS)
+Run docker ps | grep apls  -> If apls container is not running, then license server is not running, and you need to start it 
+<your install directory>/ swarm-learning/bin/run-apls 
+
+###### Check APLS status:  
+Browse to autopass license server management console: https://192.168.1.102:5814/autopass
```
Comment (suresh-ls) [inline docs/Troubleshooting.md:26]: We need to detail how to launch the GUI.. Please refer Issue #5

File: docs/FAQ.md (hunk 1 above)
Comment (RadhakrishnaJ) [reply docs/FAQ.md]: Corrected wording.

**Review by suresh-ls — APPROVED** [review]

File: docs/Troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+   
+   
+#### Resolution
+Fix for above issue involves two things.
+1.	Running AutoPass License Server (APLS) 
+2.	Setting up Swarm License
+
+##### 1. Running AutoPass License Server (APLS)
+Run docker ps | grep apls  -> If apls container is not running, then license server is not running, and you need to start it using  
+'your install directory/swarm-learning/bin/run-apls'
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:23]: Is this the convention that we are using now?
In the original version of the documentation, we had said that we were assuming the install directory to be swarm-learning or something like that. Have we removed this?
I would prefer swarm-learning-install-dir to your-install-dir. And, please format code and artifacts as code blocks.

File: docs/Troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+#### Resolution
+Fix for above issue involves two things.
+1.	Running AutoPass License Server (APLS) 
+2.	Setting up Swarm License
+
+##### 1. Running AutoPass License Server (APLS)
+Run docker ps | grep apls  -> If apls container is not running, then license server is not running, and you need to start it using  
+'your install directory/swarm-learning/bin/run-apls' 
+
+###### Check APLS status:
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:25]: Did not follow. What do you mean by "APLS status" here? Did that docker ps command above not check status?

File: docs/Troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+2.	Setting up Swarm License
+
+##### 1. Running AutoPass License Server (APLS)
+Run docker ps | grep apls  -> If apls container is not running, then license server is not running, and you need to start it using  
+'your install directory/swarm-learning/bin/run-apls' 
+
+###### Check APLS status:  
+Follow the steps mentioned below to verify APLS status. 
+
+> NOTE: For this explanation IP 192.168.1.102 is used – replace this IP address with your license host system IP address (Do NOT use localhost / 127.0.0.1)
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:28]: "license host system" should be rephrased. How about "the host system on which the license server container is running"?

File: docs/Troubleshooting.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+Run docker ps | grep apls  -> If apls container is not running, then license server is not running, and you need to start it using  
+'your install directory/swarm-learning/bin/run-apls' 
+
+###### Check APLS status:  
+Follow the steps mentioned below to verify APLS status. 
+
+> NOTE: For this explanation IP 192.168.1.102 is used – replace this IP address with your license host system IP address (Do NOT use localhost / 127.0.0.1)  
+> 5814 is the default port used for APLS. 
+   
+Browse to autopass license server management console: https://192.168.1.102:5814/autopass.
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:31]: "Open https://192.168.1.102:5814/autopass from a web browser"?
"Use a web browser to connect to https://192.168.1.102:5814/autopass"?
And, use either APLS or **A**utopass **L**icense **S**erver

File: docs/Troubleshooting.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+> NOTE: If you can run a browser on the host machine where you ran the "run-apls" script, You can now connect to the License server using https://192.168.1.102:5814/autopass from a browser.
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:32]: Did not understand this

File: docs/Troubleshooting.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+> Else, if the host machine is accessible from your laptop/desktop, then you can access by using https://192.168.1.102:5814/autopass from your laptop/desktop browser.
+> 
+> Else you should setup SSH port forwarding first and then access https://localhost:5814/autopass from your laptop/desktop browser. 
+> 
+> Run this shell command from your laptop/desktop to do SSH port forwarding  
+> - $ ssh -L 5814:192.168.1.102:5814 username@192.168.1.102 
+
+If the browser response is, "this site can’t be reached" / "refused to connect" - It means APLS is not running correctly. Restart run-apls script as mentioned above. 
+
+Otherwise, In the APLS login screen, login using "admin/password".
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:43]: Why do we need all these details? If at all it is relevant, it should go into other sections like Installation or Configuration. Have pointers to those sections here.
If the browser cannot connect, it does not necessarily mean that APLS is not running &mdash; for instance, he could have an incorrect network proxy configuration. Preferrably, we should not make such assumptions and presumptions. Instead, let him work with his admins and ensure connectivity.
That APLS password is the initial password &mdash; it is not valid for subsequent logins.

File: docs/Troubleshooting.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+Select License file (.DAT).
+
+Select all feature IDs and install. 
+
+###### Confirm License setup is successful: 
+Refer License installed picture here.
+
+   ![License_server_after_installing_license](./images/APLS_after_installing_license.png)
+ 
+   If you are seeing the above screenshot in your APLS Management console, it means your license setup is complete.
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:69]: Again, do we really require so much handholding here?

File: docs/FAQ.md [hunk 11]
```diff
@@ -88,7 +88,7 @@ The ML program can produce additional log output. To do so, it should be modifie
 
 ## Do you need sudo/root privileges to run Swarm Learning?
 
-sudo is not required to launch the container, if docker is configured to run as a [non-root user](URL.md#16-httpsdocsdockercomengineinstalllinux-postinstallmanage-docker-as-a-non-root-user).
+sudo is not required to launch the container, if docker is configured to run as a non-root user. Refer [Manage Docker as a non-root user](URL.md#16-manage-docker-as-a-non-root-user-httpsdocsdockercomengineinstalllinux-postinstallmanage-docker-as-a-non-root-user)
```
Comment (ravi-ss) [inline docs/FAQ.md:91]: > if docker is configured to run as a non-root user
Is this correct? What do you mean by "docker" here? The docker daemon? Is it really running with non-root privileges?

File: docs/FAQ.md [hunk 12]
```diff
@@ -130,8 +130,16 @@ We support the following basic packages required for ML - numpy, scipy, matplotl
 
 ## Can one add new python packages in Swarm Learning?
```
Comment (ravi-ss) [inline docs/FAQ.md:131]: Should it not be "Python"?

File: docs/FAQ.md [hunk 13]
```diff
@@ -130,8 +130,16 @@ We support the following basic packages required for ML - numpy, scipy, matplotl
...
+Yes. For any additional support packages to machine learning process, user has to do following.
+
+1. Write a dockerfile to create a local SL image
+    - By extending FROM the default SL image (sl-tf or sl-pyt).
+    - Add required additional packages
+2. Build the local SL image from above dockerfile.
+3. If local SL image name differs from default SL image, then update ‘common’ script (inside swarm-learning/bin) and make sure following lines uses local SL image. 
+    - slTFImage="${swarmDockerHubAndUser}/sl-tf:${swarmVer}" (or)
+    - slPytImage="${swarmDockerHubAndUser}/sl-pyt:${swarmVer}"
+4. As usual run the container using run-sl command.
```
Comment (ravi-ss) [inline docs/FAQ.md:142]: To the extent possible, provide exact commands and line numbers. Readers frequently copy-paste these instructions and get surprised when it does not work.

File: docs/setup.md [hunk 14]
```diff
@@ -9,7 +9,7 @@ The evaluation license for running the Swarm Learning components are available a
 ## Pulling docker images
 
 Swarm Learning docker images are available on HPE docker registry - ``hub.myenterpriselicense.hpe.com.`` Docker registry access is password
-protected. User needs to login to docker registry using HPE Passport email id and password 'hpe_eval'. All Swarm Learning images are signed by HPE with the name as 'hpe-ai-swarm-learning'. Users can inspect and pull these images after enabling docker content trust by setting the environment variable as 'DOCKER_CONTENT_TRUST=1'. See [resources](URL.md#21-httpsdocsdockercomenginesecuritytrust) for details.
+protected. User needs to login to docker registry using HPE Passport email id and password 'hpe_eval'. All Swarm Learning images are signed by HPE with the name as 'hpe-ai-swarm-learning'. Users can inspect and pull these images after enabling docker content trust by setting the environment variable as 'DOCKER_CONTENT_TRUST=1'. See [Content trust in Docker](URL.md#21-content-trust-in-docker-httpsdocsdockercomenginesecuritytrust) for details.
```
Comment (ravi-ss) [inline docs/setup.md:12]: So, when I click that link, I see 20 references. Which is the relevant one?

File: docs/setup.md [hunk 15]
```diff
@@ -82,7 +82,7 @@ Licenses for running the Swarm Learning package should be installed on the Licen
 
 The default username is, ``"admin"``, and the password, ``"password"``.
```
Comment (ravi-ss) [inline docs/setup.md:83]: Why those double quotes? It prevents readers from copy-pasting.

File: docs/setup.md [hunk 16]
```diff
@@ -82,7 +82,7 @@ Licenses for running the Swarm Learning package should be installed on the Licen
...
 
-Use the management interface to install licenses and manage them. See chapter "HPE AutoPass License Server License Management" in [AutoPass License Server User Guide](HPE%20AutoPass%20License%20Server%20User%20Guide.pdf).
+Use the management interface to install licenses and manage them. See chapter "HPE AutoPass License Server License Management" in [AutoPass License Server User Guide](HPE%20AutoPass%20License%20Server%20User%20Guide.pdf) for details of the web GUI and how to install license.
 
 > NOTE: Whenever the container running the License Server (apls) is stopped or the host system running the License Sever reboots, the  License Server needs to be started and the licenses have to be reinstalled again.
```
Comment (ravi-ss) [inline docs/setup.md:87]: Should be **APLS**, not _apls_.

File: docs/setup.md [hunk 17]
```diff
@@ -82,7 +82,7 @@ Licenses for running the Swarm Learning package should be installed on the Licen
 
 The default username is, ``"admin"``, and the password, ``"password"``.
 
-Use the management interface to install licenses and manage them. See chapter "HPE AutoPass License Server License Management" in [AutoPass License Server User Guide](HPE%20AutoPass%20License%20Server%20User%20Guide.pdf).
+Use the management interface to install licenses and manage them. See chapter "HPE AutoPass License Server License Management" in [AutoPass License Server User Guide](HPE%20AutoPass%20License%20Server%20User%20Guide.pdf) for details of the web GUI and how to install license.
 
 > NOTE: Whenever the container running the License Server (apls) is stopped or the host system running the License Sever reboots, the  License Server needs to be started and the licenses have to be reinstalled again. 
 > HPE recommends not to stop the License Server container once it is started.
```
Comment (ravi-ss) [inline docs/setup.md:88]: Can you make this stronger? Recommendations can be ignored &mdash; and, generally speaking, I would expect a product to work properly even when I ignore a recommendation. Our case is different &mdash; we simply do not want users to stop APLS.

File: examples/fraud-detection-keras/README.md [hunk 18]
```diff
@@ -12,7 +12,7 @@ Scripts to run the example are in the ``swarm-learning/examples/fraud-detection-
 - del-workspace: It deletes the created workspace and also stops all containers.
   
 Following environment variables are required to set in each terminal:
--	APLS_IP: IP address of the host where license server is running. 
+-	APLS_IP: IP address of the host where license server is running (Do NOT use localhost / 127.0.0.1).
```
Comment (ravi-ss) [inline examples/fraud-detection-keras/README.md:15]: External IP?

File: examples/fraud-detection-keras/README.md [hunk 19]
```diff
@@ -23,7 +23,7 @@ Before running the example, verify license server is running and valid license i
 #### 1.	Create workspace and initialize Swarm Learning 
 First open a terminal, change directory to ``swarm-learning/examples`` and set the following environment variables. Then run ``init-workspace`` script to create workspace for ``fraud-detection-keras`` example and initialize Swarm Learning environment by running Swarm Network and Spire Server containers.
 
-   ``APLS_IP=<License Server IP>``
+   ``APLS_IP=<License Host Server IP (Do NOT use localhost / 127.0.0.1)>``
```
Comment (ravi-ss) [inline examples/fraud-detection-keras/README.md:26]: This is applicable for any and every IP the user specifies. Can we have a single, big warning somewhere instead of reiterating all over the place?

**Review by ravi-ss — CHANGES_REQUESTED** [review]

File: docs/Troubleshooting.md [hunk 20]
```diff
@@ -0,0 +1,69 @@
+## Troubleshooting ##
+
+Troubleshooting provides solutions to commonly observed issues during Swarm set up and execution.
+
+
+### 1. Error code : 6002
+####    Error message: Unable to connect to server. Server might be wrongly configured or down. #### 
+####    Custom message: Error in communicating with server https://HOST_SYSTEM_IP:5814 (default port) ####
```
[the remark this answers is not in the document]
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md:8]: Formatted Error messages as code block.

File: docs/Troubleshooting.md [hunk 21]
```diff
@@ -0,0 +1,69 @@
...
+   ![Error_6002](./images/Error_6002_running_SL_node_without_Server_and_License.png)
+   
+   
+#### Resolution
+Fix for above issue involves two things.
+1.	Running AutoPass License Server (APLS) 
+2.	Setting up Swarm License
+
+##### 1. Running AutoPass License Server (APLS)
+Run docker ps | grep apls  -> If apls container is not running, then license server is not running, and you need to start it using
```
[the remark this answers is not in the document]
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md:22]: Formatted as code block. If APLS name changes, this section needs update.

File: docs/Troubleshooting.md (hunk 4 above)
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md:23]: Corrected swarm-learning-install-dir and formatted as code block.

File: docs/Troubleshooting.md (hunk 6 above)
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md:28]: corrected wording

File: docs/Troubleshooting.md (hunk 7 above)
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md:31]: Corrected wording and used APLS.

File: docs/Troubleshooting.md (hunk 9 above)
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md:43]: Most of the time license server issues are first time user issues, once first time is setup they will good. Detailed instructions are added to support first time user.

File: docs/Troubleshooting.md (hunk 10 above)
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md:69]: Seeing the pattern of license server issues and explanation, decided to have screen shots and detailed instructions to help first time users.

File: docs/FAQ.md (hunk 11 above)
Comment (RadhakrishnaJ) [reply docs/FAQ.md:91]: Corrected reference link to point the resources. Corrected name to match as in references section, so that it is easy for user get docker documentation.

File: docs/FAQ.md (hunk 12 above)
Comment (RadhakrishnaJ) [reply docs/FAQ.md:131]: Done

File: docs/FAQ.md (hunk 13 above)
Comment (RadhakrishnaJ) [reply docs/FAQ.md:142]: Scripts can change frequently, may be giving lines needs frequent revisit of this. These instructions are not to be used as is, user needs to modify to match with his local image.

File: docs/setup.md (hunk 14 above)
Comment (RadhakrishnaJ) [reply docs/setup.md:12]: Earlier one was going to resources main heading. Now it is going to page where relevant link is available, I have maintained name to match with name in references, using the name user has to refer the link. Didn't find any other better way.

File: docs/setup.md (hunk 15 above)
Comment (RadhakrishnaJ) [reply docs/setup.md:83]: Done

File: docs/setup.md (hunk 16 above)
Comment (RadhakrishnaJ) [reply docs/setup.md:87]: Done

File: docs/setup.md (hunk 17 above)
Comment (RadhakrishnaJ) [reply docs/setup.md:88]: Moved it out of Notes. Made it little bold with small font.

File: examples/fraud-detection-keras/README.md (hunk 18 above)
Comment (RadhakrishnaJ) [reply examples/fraud-detection-keras/README.md:15]: No update done for this. Browsed for external IP, it has multiple definitions.

File: examples/fraud-detection-keras/README.md (hunk 19 above)
Comment (RadhakrishnaJ) [reply examples/fraud-detection-keras/README.md]: In Examples - removed repeated occurrences of Do NOT use localhost / 127.0.0.1.

File: docs/Troubleshooting.md (hunk 9 above)
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md]: Moved these steps from Troubleshooting to setup. Added link in troubleshooting to setup.

File: docs/Troubleshooting.md (hunk 10 above)
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md]: Moved content to Setup documentation. Only one screen shot used to show how management console looks after successful license install.

**Comment by RadhakrishnaJ** [comment]
Completed rework. Please perform review.

File: docs/setup.md [hunk 22]
```diff
@@ -78,14 +78,68 @@ The steps to be followed to download Swarm Learning docker images on the Linux e
 
 ## Installing licenses and starting license server
 
-Licenses for running the Swarm Learning package should be installed on the License Server. After completing the product installation, start the License Server container using the ``swarm-learning/bin/run-apls`` script. The parameters for this script are described in the Running Swarm Learning section. Then, connect to the License Server's management interface using a web browser. This site can be reached using the link, ``-address-or-FQDN-of-host-running-License-Server-container>:5814/autopass``.
+#### Running AutoPass License Server (APLS)
```
Comment (suresh-ls) [inline docs/setup.md:81]: Looks too cluttered :-)   Can we number them ?  Say 1. Running AutoPass....     2) Setting up Swarm License  at the very least...

**Review by suresh-ls — APPROVED** [review]

File: docs/Troubleshooting.md (hunk 8 above)
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md]: Added steps to connect with APLS GUI.

File: docs/setup.md (hunk 22 above)
Comment (RadhakrishnaJ) [reply docs/setup.md:81]: Will handle adding bullets in next pull.

File: docs/Troubleshooting.md [hunk 23]
```diff
@@ -0,0 +1,31 @@
+## Troubleshooting ##
+
+Troubleshooting provides solutions to commonly observed issues during Swarm set up and execution.
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:3]: Please use **Swarm Learning** everywhere &mdash; not Swarm

File: docs/Troubleshooting.md [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+
+
+### 1. Error code : 6002
+####    ``Error message: Unable to connect to server. Server might be wrongly configured or down.`` #### 
+####    ``Custom message: Error in communicating with server https://HOST_SYSTEM_IP:5814 (default port)`` #### 
+
+#### Problem Description 
+Error code: 6002, as shown in below screenshot happens when SL is not able to connect to the APLS server.
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:11]: SL = Swarm Learning?
I prefer you rephrase this as "Swarm Learning components" or something similar. SL is confusing because we also use it to refer to the ML nodes.

File: docs/Troubleshooting.md [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+Error code: 6002, as shown in below screenshot happens when SL is not able to connect to the APLS server. 
+
+   ![Error_6002](./images/Error_6002_running_SL_node_without_Server_and_License.png)
+   
+   
+#### Resolution
+Fix for above issue requires to make sure following two things.
+
+##### 1. Running AutoPass License Server (APLS)
+Run ``docker ps | grep apls``  -> If apls container is not running, then license server is not running, and you need to start it using
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:20]: Where is this **apls** name hardcoded? It is not in `run-apls`.

I repeat again:
1. Do not provide instructions that are definitely wrong.
2. To the extent possible, provide exact commands &mdash; commands that people can copy-paste.

So, for this specific case:
1. Just say, "Run `docker ps` and check whether APLS is running". Do not include the `grep`.
2. If you really want to filter the results, use a filter on the image, not container. See `genDockerExec` in `common`: `docker ps --format {{.Names}} --filter=ancestor=hub.myenterpriselicense.hpe.com/hpe_eval/swarm-learning/apls:0.3.0`

File: docs/Troubleshooting.md [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+#### Problem Description 
+Error code: 6002, as shown in below screenshot happens when SL is not able to connect to the APLS server. 
+
+   ![Error_6002](./images/Error_6002_running_SL_node_without_Server_and_License.png)
+   
+   
+#### Resolution
+Fix for above issue requires to make sure following two things.
+
+##### 1. Running AutoPass License Server (APLS)
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:19]: Verify APLS is running

File: docs/Troubleshooting.md [hunk 27]
```diff
@@ -0,0 +1,31 @@
...
+Run ``docker ps | grep apls``  -> If apls container is not running, then license server is not running, and you need to start it using  
+``swarm-learning-install-dir/swarm-learning/bin/run-apls`` 
+
+Browse to APLS management console.
+If the browser response is, "this site can’t be reached" / "refused to connect" - It means APLS is not running correctly. Restart run-apls script as mentioned above.
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:24]: If the browser cannot connect, verify the network proxy settings, firewall policies, etc. that are in effect. If required, work with your network administrator to resolve them.

Do not advice restarting APLS. We definitely discourage that.

File: docs/setup.md [hunk 28]
```diff
@@ -78,14 +78,68 @@ The steps to be followed to download Swarm Learning docker images on the Linux e
...
+ Start the APLS container using
+``swarm-learning-install-dir/swarm-learning/bin/run-apls`` 
 
-The default username is, ``"admin"``, and the password, ``"password"``.
+###### Browse APLS Management console:
```
Comment (ravi-ss) [inline docs/setup.md:85]: "Access" or "Connect to" instead of "browse"?

File: docs/setup.md [hunk 29]
```diff
@@ -78,14 +78,68 @@ The steps to be followed to download Swarm Learning docker images on the Linux e
...
-Licenses for running the Swarm Learning package should be installed on the License Server. After completing the product installation, start the License Server container using the ``swarm-learning/bin/run-apls`` script. The parameters for this script are described in the Running Swarm Learning section. Then, connect to the License Server's management interface using a web browser. This site can be reached using the link, ``-address-or-FQDN-of-host-running-License-Server-container>:5814/autopass``.
+#### Running AutoPass License Server (APLS)
+ Start the APLS container using
+``swarm-learning-install-dir/swarm-learning/bin/run-apls`` 
 
-The default username is, ``"admin"``, and the password, ``"password"``.
+###### Browse APLS Management console:  
+Follow the steps mentioned below to browse APLS management console. 
+
+> NOTE: For this explanation IP 192.168.1.102 is used – replace this IP address with the host system IP address on which the license server container is running.
```
Comment (ravi-ss) [inline docs/setup.md:88]: Should we use the same IP that we have in our examples?

File: docs/setup.md [hunk 30]
```diff
@@ -78,14 +78,68 @@ The steps to be followed to download Swarm Learning docker images on the Linux e
...
+- If you can run a browser on the host machine where you ran the "run-apls" script, You can now connect to the License server using https://192.168.1.102:5814/autopass from a browser. 
+ 
+- Else, if the host machine is accessible from your laptop/desktop, then you can access by using https://192.168.1.102:5814/autopass from your laptop/desktop browser.
+
+- Else you should setup SSH port forwarding first and then access https://localhost:5814/autopass from your laptop/desktop browser. 
+
+> NOTE : Running below shell command is one of the way to do SSH port forwarding from your laptop/desktop.
+> - $ ssh -L 5814:192.168.1.102:5814 username@192.168.1.102 
+
+If the browser response is, "this site can’t be reached" / "refused to connect" - It means APLS is not running correctly. Restart run-apls script as mentioned above.
```
Comment (ravi-ss) [inline docs/setup.md:103]: How about something on these lines?

Access the License Server's management console by using a browser to connect to https://192.160.1.102:5814/.

NOTE:
1. These instructions assume the IP of the system on which the License Server is running is 192.168.1.102.
2. They also assume the container port has been published and that the external port is 5814.
3. Modify these values to match the actual IP and external port on your system.
4. If the browser cannot connect to the management console, check your network proxy settings and firewall policies. Consider techniques like port forwarding to workaround firewall policies. If necessary, work with your network administrator to diagnose and resolve connectivity problems.

**Review by ravi-ss — CHANGES_REQUESTED** [review]

File: docs/Troubleshooting.md (hunk 25 above)
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md]: updated to use only docker ps.

File: docs/Troubleshooting.md (hunk 27 above)
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md]: Added this statement.

File: docs/setup.md (hunk 29 above)
Comment (RadhakrishnaJ) [reply docs/setup.md:88]: I don't see any APLS host machine IP in our examples. As explanation is given the used IP, I hope this is good enough.

File: docs/setup.md (hunk 28 above)
Comment (RadhakrishnaJ) [reply docs/setup.md:85]: used Access

File: docs/setup.md (hunk 30 above)
Comment (RadhakrishnaJ) [reply docs/setup.md]: Updated documentation.

File: docs/Troubleshooting.md [hunk 31]
```diff
@@ -0,0 +1,32 @@
...
+   
+#### Resolution
+Fix for above issue requires to make sure following two things.
+
+##### 1. Verify APLS is running 
+Run ``docker ps`` and check whether APLS is running. If apls container is not running, then license server is not running, and you need to start it using  
+``swarm-learning-install-dir/swarm-learning/bin/run-apls`` 
+
+Access APLS management console.
+If the browser response is, "this site can’t be reached" / "refused to connect".
```
Comment (ravi-ss) [inline docs/Troubleshooting.md:24]: Delete this

**Review by ravi-ss — APPROVED** [review]

File: docs/Troubleshooting.md (hunk 31 above)
Comment (RadhakrishnaJ) [reply docs/Troubleshooting.md]: Done
